Source/Krypton Components/Krypton.Toolkit/Controls Toolkit/KryptonCheckedListBox.cs

```diff
@@ -2404,6 +2518,53 @@ private void OnListBoxMouseChange(object? sender, EventArgs e)
                 }
             }
         }
+
+        private void RefreshItems()
+        {
+            _checkedListBox.BeginUpdate();
+
+            _checkedListBox.Items.Clear();
+
+            if (_dataSource != null)
+            {
+                if (BindingContext != null && BindingContext[_dataSource] is CurrencyManager cm)
+                {
+                    for (int i = 0; i < cm!.Count; i++)
```

Comment:
Why is this null thing here, it should not be needed as it will not pass in because of the `is` clause above

Reply:
And should not the `i` be related to the `List size` and not what count of the base is set to ?

Reply:
The code has been changed since